### PR TITLE
SCNINE-2791 | Financials/Composites/xxx/Accounts - Associations Filter broken - replace with Dropdown

### PR DESCRIPTION
**Pull request by gstrauss-confluence**
- [ ] SCNINE-2791

### CODE REVIEW

File: lib/data_table.rb [hunk 1]
```diff
@@ -247,9 +247,9 @@ def tables
       attr_accessor :sorted_and_filtered_query
       def query!(params)
         @sorted_and_filtered_query = @base_query.arel.dup # convert to AREL
+        apply_grouping!(params)
```
Comment (WillemOpperman) [inline lib/data_table.rb:250]: you need to rebase on `statpro/devextreme-rails` master

File: lib/data_table.rb [hunk 2]
```diff
@@ -285,6 +285,7 @@ def query!(params)
 
         group_params = JSON.parse(group_params) if group_params.is_a? String
 
+        @sorted_and_filtered_query.orders.clear
```
Comment (WillemOpperman) [inline lib/data_table.rb:288]: you need to rebase on `statpro/devextreme-rails` master

File: lib/data_table.rb [hunk 3]
```diff
@@ -760,19 +760,16 @@ def data_field_for(name)
...
-          Jbuilder.encode do |json|
-
-            json.items(resultset) do |instance|
-
+        group_params = params.fetch('groupOptions', {})
+        group_params = JSON.parse(group_params) if group_params.is_a?(String) && group_params.present?
+
+        Jbuilder.encode do |json|
+          json.items(resultset) do |instance|
+            if !params.key?(:skip) && !params.key?(:take) && group_params.size == 1 # if lookup filter row
```
Comment (ChristopherRollins) [inline lib/data_table.rb:768]: Please add a comment explaining what this is for and why it is here. Also, why are you using `skip` and `take`? Is that the only way to identify that this is from the filter row?

Also, I don't like the fact that the params check is inside the loop. This adds unnecessary compute to this loop. Please move those outside the loop. Duplicate if need be.

**Review by ChristopherRollins — DISMISSED** [review]

File: lib/data_table.rb [hunk 4]
```diff
@@ -802,9 +816,9 @@ def to_json(view_context, params = {})
                 actions = self.actions.map { |action| build_action(action, instance, view_context) }.reject(&:blank?)
                 json._actions actions.to_json
               end
-            end
 
-            json.total_count(total_count) if total_count
+              json.total_count(total_count) if total_count
```
Comment (ChristopherRollins) [inline lib/data_table.rb:820]: Why is this in the array now?

File: lib/data_table.rb [hunk 5]
```diff
@@ -760,11 +760,23 @@ def data_field_for(name)
...
+
+        # If the request is for a lookup column or from headerfilter(Date)
+        #   set the appropriate column value for the lookup column and headerfilter(Date) such that the correct values are displayed in the dropdown
+        # else
+        #   default
+        if column_selector.present?
           Jbuilder.encode do |json|
             json.items(resultset) do |instance|
-              value = request_column.text(instance, view_context) rescue nil
+              value = self.columns.detect{ |c| c.name == column_selector }.text(instance, view_context) rescue nil
```
Comment (ChristopherRollins) [inline lib/data_table.rb:779]: Please move the detect outside of this array. You can assign it just after the if condition. Like it was being done before.
ie.
```
        if column_selector.present?
          request_column = self.columns.detect{ |c| c.name == column_selector }
          Jbuilder.encode do |json|
            json.items(resultset) do |instance|
              value = request_column.text(instance, view_context) rescue nil
```

**Review by WillemOpperman — APPROVED** [review]

**Pull request merged**